src/winkerberos.c

```diff
@@ -437,8 +437,160 @@ sspi_client_clean(PyObject* self, PyObject* args) {
+    char *acceptor_address = NULL;
+    char *application_data = NULL;
+
+    size_t offset = 0;
+    size_t data_length = 0;
+    unsigned char *offset_p = NULL;
+
+    if (!PyArg_ParseTupleAndKeywords(args, keywds, "|ls#ls#s#", kwlist,
+            &initiator_addrtype, &initiator_address, &initiator_length,
+            &acceptor_addrtype, &acceptor_address, &acceptor_length,
+            &application_data, &application_length)) {
+        return NULL;
+    }
+
+    data_length = initiator_length + acceptor_length + application_length;
+    offset = sizeof(SEC_CHANNEL_BINDINGS);
+
+    context_bindings = (SecPkgContext_Bindings *) malloc(sizeof(SecPkgContext_Bindings));
+    if (context_bindings == NULL) {
+        PyErr_SetString(PyExc_MemoryError, "Failed to allocate a memory block for SecPkgContext_Bindings");
+        return NULL;
+    }
+    context_bindings->BindingsLength = sizeof(SEC_CHANNEL_BINDINGS) + data_length;
+
+    channel_bindings = (SEC_CHANNEL_BINDINGS*) malloc(context_bindings->BindingsLength);
+    if (channel_bindings == NULL) {
+        free(context_bindings);
+        PyErr_SetString(PyExc_MemoryError, "Failed to allocate a memory block for SEC_CHANNEL_BINDINGS");
+        return NULL;
+    }
+    context_bindings->Bindings = channel_bindings;
+
+    channel_bindings->dwInitiatorAddrType = initiator_addrtype;
+    channel_bindings->cbInitiatorLength = initiator_length;
+    if (initiator_address != NULL) {
+        channel_bindings->dwInitiatorOffset = offset;
+
+        offset_p = &((unsigned char*) channel_bindings)[offset];
+        memcpy_s(&offset_p[0], initiator_length, initiator_address, initiator_length);
+        offset = offset + initiator_length;
+    } else {
+        channel_bindings->dwInitiatorOffset = 0;
+    }
+
+    channel_bindings->dwAcceptorAddrType = acceptor_addrtype;
+    channel_bindings->cbAcceptorLength = acceptor_length;
+    if (acceptor_address != NULL) {
+        channel_bindings->dwAcceptorOffset = offset;
+
+        offset_p = &((unsigned char*) channel_bindings)[offset];
+        memcpy_s(&offset_p[0], acceptor_length, acceptor_address, acceptor_length);
+        offset = offset + acceptor_length;
+    } else {
+        channel_bindings->dwAcceptorOffset = 0;
+    }
+
+    channel_bindings->cbApplicationDataLength = application_length;
+    if (application_data != NULL) {
+        channel_bindings->dwApplicationDataOffset = offset;
+        offset_p = &((unsigned char*) channel_bindings)[offset];
+        memcpy_s(&offset_p[0], application_length, application_data, application_length);
+    } else {
+        channel_bindings->dwApplicationDataOffset = 0;
+    }
+
+    pychan_bindings = PyCObject_FromVoidPtr(context_bindings, &destruct_channel_bindings_struct);
+    if (pychan_bindings == NULL) {
+        free(channel_bindings);
+        free(context_bindings);
+        PyErr_SetString(PyExc_RuntimeError, "NULL result returned when creating PyCObject of a SecPkgContext_Bindings struct");
```

Comment:
If PyCapsule_New or PyCObject_FromVoidPtr fail an exception with more detail will automatically be set. You shouldn't set RuntimeError here, just return NULL.